### PR TITLE
Fix: audio encoder uses full attention instead of windowed on non-FA2 backends

### PR DESCRIPTION
**Pull request by Dvad**
**Bug:** Qwen3ASRAudioEncoder.forward() passes `cu_seqlens` to attention layers but never builds a 4D attention mask. Only Flash Attention 2 (CUDA) interprets cu_seqlens for windowed attention boundaries. 

On SDPA/eager backends (MPS, CPU), cu_seqlens is ignored and the encoder performs full global self-attention over all tokens instead of the trained windowed attention pattern.

This causes significant quality degradation on non-CUDA hardware (~340 words transcribed vs ~555 expected on a 5-minute test clip).

  Fix: Call the existing _prepare_attention_mask() method (which already returns None for FA2) and pass the resulting block-diagonal mask to each encoder layer.
```
           cu_seqlens = torch.tensor(cu_chunk_lens, device=aftercnn_lens.device).cumsum(-1, dtype=torch.int32)
  +        attention_mask = self._prepare_attention_mask(hidden_states, cu_seqlens)
           for encoder_layer in self.layers:
               layer_outputs = encoder_layer(
                   hidden_states,
                   cu_seqlens,
  +                attention_mask=attention_mask,
               )
```

Verified on: MPS (Apple Silicon), CPU.

FA2 path is unchanged (`_prepare_attention_mask` returns `None` for flash_attention_2).

### CODE REVIEW

**Comment by moona3k** [comment]
Corroborating this with measurements (see #213 for the full writeup). At `7c6daf7`, CPU fp32 `sdpa`, torch 2.10.0, `Qwen/Qwen3-ASR-0.6B`:

- Clips <= 8 s (one `n_window_infer` window): encoder output unchanged by this patch, as expected.
- 20 s clip: as-shipped vs this patch, mean|diff| = 0.0025, max = 0.0955, relative = 18.5%. Across 17 LibriSpeech clips > 8 s the relative divergence was 4.7-12.8% and grows with duration.
- This patch's dense block-diagonal mask is bit-identical (max |diff| = 0.0) to splitting Q/K/V at `cu_seqlens` and attending per window, which is what `transformers` `main` and the vLLM backend here do. So the fix is exact, not an approximation.

The change is correct and minimal; the only nit is trailing whitespace on the three added lines.

**Comment by Mark-Oliass** [comment]
Supporting this. We independently arrived at the same patch (build the mask once outside the loop,
pass it to every layer) before finding this PR, and it fixed a severe production failure for us:
with `sdpa` the audio encoder's window was not applied, and greedy decoding fell into repetition
loops on 37.5 % of speech blocks. With this patch that drops to 14.3 %, and to 0 on audio that is
actually transcribable — matching the vLLM backend on the same audio.

Details and measurements are in #213. One data point for the review, if it helps: on our input this
block-diagonal mask is *exactly* equivalent to actually running flash-attn (identical merged text,
per-chunk text, and per-chunk token counts), so the "approximation" caveat in the docstring does
not show up here.
